website/pages/component/[id]/index.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { GetStaticPaths, GetStaticProps } from 'next';
+import fs from 'fs';
+import path from 'path';
+import Layout from '../../../components/layout';
+import { BaseLayoutProps, markdownToHtml } from '../../../utils';
+
+export default function Component({
+  component,
+  changelog,
+  readme,
+}: BaseLayoutProps) {
+  return <Layout component={component} changelog={changelog} readme={readme} />;
+}
+
+export const getStaticProps: GetStaticProps = async ({ params }) => {
+  if (!params) {
+    return { props: {} };
+  }
+
+  const { id } = params;
+
+  let changelogMarkdown, readme;
+
+  if (typeof id === 'string') {
+    changelogMarkdown = fs.readFileSync(
+      path.join('../packages', id, 'changelog.md'),
+    );
+
+    readme = fs
+      .readFileSync(path.join('../packages', id, 'README.md'))
```

Comment:
Let's do this async instead so we don't lock the thread.

Reply:
If we do this async then the prop shows up in the Component component as undefined

Reply:
We have an async await below, that works fine no? May be some bug when switching to the async version.

Reply:
That stops working when the original request isn't completed before the value is needed on line 74. I can't await the readFile function, so it's trying to convert the changelog from markdown to HTML before there's a value for the changelogMarkdown.

Reply:
They do provide a promise API now: https://nodejs.org/api/fs.html#fs_promise_example

Reply:
used a utils wrapper but I can update to this if you think that's better

Reply:
Def should be async please!

Reply:
Yes made it async! By using util to turn fs response into a promise
`const getFileContent = util.promisify(fs.readFile);`